src/connectors/impls/clickhouse.rs

```diff
@@ -0,0 +1,290 @@
+        let config = raw_config
+            .config
+            .as_ref()
+            .ok_or_else(|| ErrorKind::MissingConfiguration(alias.to_string()))?;
+
+        let config = ClickhouseConfig::new(config)?;
+
+        Ok(Box::new(Clickhouse { config }))
+    }
+}
+
+pub struct Clickhouse {
+    config: ClickhouseConfig,
+}
+
+#[async_trait::async_trait]
+impl Connector for Clickhouse {
+    async fn create_sink(
+        &mut self,
+        sink_context: SinkContext,
+        builder: SinkManagerBuilder,
+    ) -> Result<Option<SinkAddr>> {
+        let db_url = self.connection_url();
+        let columns = self
+            .config
+            .columns
+            .iter()
+            .map(|Column { name, type_ }| (name.clone(), type_.clone()))
+            .collect();
+
+        let sink = ClickhouseSink {
+            db_url,
+            pool: None,
+            columns,
+        };
+        builder.spawn(sink, sink_context).map(Some)
+    }
+
+    fn codec_requirements(&self) -> CodecReq {
+        CodecReq::Structured
+    }
+}
+
+impl Clickhouse {
+    fn connection_url(&self) -> String {
+        let url = self.config.url.as_str();
+        let compression = self.config.compression.unwrap_or_default();
+
+        format!("{url}?compression={compression}")
+    }
+}
+
+#[derive(Deserialize)]
+struct ClickhouseConfig {
+    url: Url<ClickHouseDefaults>,
+    compression: Option<Compression>,
+    columns: Vec<Column>,
+}
+
+impl ConfigImpl for ClickhouseConfig {}
+
+#[derive(Clone, Copy, Deserialize)]
+#[serde(rename_all = "snake_case")]
+enum Compression {
+    None,
+    Lz4,
+}
+
+impl Default for Compression {
+    fn default() -> Compression {
+        Compression::None
+    }
+}
+
+impl Display for Compression {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        match self {
+            Compression::None => "none",
+            Compression::Lz4 => "lz4",
+        }
+        .fmt(f)
+    }
+}
+
+#[derive(Deserialize)]
+struct Column {
+    name: String,
+    #[serde(rename = "type")]
+    type_: DummySqlType,
+}
+
+pub(crate) struct ClickhouseSink {
+    db_url: String,
+    pool: Option<Pool>,
+    columns: Vec<(String, DummySqlType)>,
+}
+
+#[async_trait::async_trait]
+impl Sink for ClickhouseSink {
+    async fn connect(&mut self, _ctx: &SinkContext, _attempt: &Attempt) -> Result<bool> {
+        let pool = Pool::new(self.db_url.as_str());
+        self.pool = Some(pool);
+
+        Ok(true)
+    }
+
+    async fn on_event(
+        &mut self,
+        _input: &str,
+        event: Event,
+        _ctx: &SinkContext,
+        _serializer: &mut EventSerializer,
+        _start: u64,
+    ) -> Result<SinkReply> {
+        let mut client = self
+            .pool
+            .as_ref()
+            .ok_or_else(|| Error::from(ErrorKind::NoClickHouseClientAvailable))?
+            .get_handle()
+            .await?;
+
+        let (event_min_size, event_max_size) = event.value_iter().size_hint();
```

Comment:
you can use `event.len()` to get the correct size :)

Reply:
Just seen your implementation on size hint, this also makes a lot of sense. Lets keep it this way! :)

Reply:
This should be fixed as part of [d948725](https://github.com/tremor-rs/tremor-runtime/pull/1538/commits/d9487251771ab07d16b4caac11e9789ef8616ce0)